src/spring-cloud/azext_spring_cloud/application_configuration_service.py

```diff
@@ -0,0 +1,224 @@
+    if is_enterprise_tier(cmd, resource_group, service):
+        properties = models.ConfigurationServiceGitProperty()
+        acs_resource = models.ConfigurationServiceResource(properties=properties)
+        return client.configuration_services.begin_create_or_update(resource_group, service, acs_resource)
+    else:
+        raise CLIError("Application Configuration Service is only supported in Enterprise tier.")
+
+
+def application_configuration_service_git_add(cmd, client, service, resource_group,
+                                              name, patterns, uri, label,
+                                              search_paths=None,
+                                              username=None,
+                                              password=None,
+                                              host_key=None,
+                                              host_key_algorithm=None,
+                                              private_key=None,
+                                              strict_host_key_checking=None,
+                                              no_wait=False):
+    if patterns:
+        patterns = patterns.split(",")
+    repo = models.ConfigurationServiceGitRepository(name=name, patterns=patterns, uri=uri, label=label)
+
+    if search_paths:
+        search_paths = search_paths.split(",")
+    repo.search_paths = search_paths
+    repo.username = username
+    repo.password = password
+    repo.host_key = host_key
+    repo.host_key_algorithm = host_key_algorithm
+    repo.private_key = private_key
+    repo.strict_host_key_checking = strict_host_key_checking
+
+    acs_resource = client.configuration_services.get(resource_group, service)
+    if acs_resource is None or acs_resource.properties is None:
+        raise CLIError("Application Configuration Service is not enabled.")
+    
+    acs_settings = acs_resource.properties.settings
+
+    if acs_settings is None or acs_settings.git_property is None or acs_settings.git_property.repositories is None:
+        repos = []
+        git_property = models.ConfigurationServiceGitProperty(repositories=repos)
+        acs_settings = models.ConfigurationServiceSettings(git_property=git_property)
+    else:
+        repos = acs_settings.git_property.repositories
+    
+    repos.append(repo)
+    acs_settings.git_property.repositories = repos
+    acs_resource.properties.settings = acs_settings
+
+    logger.warning("[1/2] Validating Application Configuration Service settings")
+    _validate_acs_settings(client, resource_group, service, acs_settings)
+
+    logger.warning("[2/2] Updating Application Configuration Service settings, (this operation can take a while to complete)")
+    return sdk_no_wait(no_wait, client.configuration_services.begin_create_or_update, resource_group, service, acs_resource)
+
+
+def application_configuration_service_git_update(cmd, client, service, resource_group, name,

```

Comment:
update, add and remove have simliar steps:
1. Get the list of repos
2. Find the target repo or throw exception according to the found/not found result
2. set the new repo in the repos list
3. update asc resource.
Do u wanna extract the common function to shorten the duplicated lines?